src/plugins/discover/public/application/main/components/top_nav/discover_topnav.tsx

```diff
@@ -203,7 +214,7 @@ export const DiscoverTopNav = ({
     <AggregateQueryTopNavMenu
       appName="discover"
       config={topNavMenu}
-      indexPatterns={[dataView]}
+      indexPatterns={dataViewList}
```

Comment:
It's nice that the filter is editable now but `Warning` message is missing. I think it would be very convenient for users to see which filters don't allow them to view data (by noticing Warning badge). Also Lens shows the warning.

Before:
<img width="300" alt="Screenshot 2023-07-03 at 11 17 03" src="https://github.com/elastic/kibana/assets/1415710/fc529ba4-e403-4598-b0c7-946dea719231">

With PR changes:
<img width="300" alt="Screenshot 2023-07-03 at 11 16 34" src="https://github.com/elastic/kibana/assets/1415710/f51be91e-0bfa-4c30-9b61-ae8b944bdada">

Would it be possible to keep `Warning` if the filter's data view does not match the current data view?

Reply:
Good point, about the code checking for the Warning

https://github.com/elastic/kibana/blob/ad85cc0727d1009eebcfb5ca4eab23adeb19c91b/src/plugins/unified_search/public/filter_bar/filter_item/filter_item.tsx#L282-L299

https://github.com/elastic/kibana/blob/ad85cc0727d1009eebcfb5ca4eab23adeb19c91b/src/plugins/unified_search/public/filter_bar/filter_item/filter_item.tsx#L329-L337

For me it looks it's code that's for the use case that supports multiple data views, like Lens + Dashboard. In Discover we just allow a single data view to be selected. So it might be worth not changing the code on Discover side, but to better support this case in the Filter UI, like, telling the some details of the warning (Filter was created with a different data view that's not available now). WDYT @lukasolson

Reply:
Hmm, what I don't currently like about this implementation is that as soon as you refresh the page, you'll get the warning again since this relies on changes in state to get the right value for `dataViewList`.

The data views list that gets passed to the filter bar currently represents the data views that are applicable to the current view. This is also the case currently for Lens and Dashboard, which also have this "bug".

I agree it's somewhat of an inconvenience that you have to select a different data view in order to even edit the filter... Maybe we can have a quick chat to align on what the best behavior here should be.